internal/pkg/cli/app_init.go

```diff
@@ -83,9 +83,13 @@ func (opts *InitAppOpts) Validate() error {
 		}
 	}
 	if opts.DockerfilePath != "" {
-		if _, err := listDockerfiles(opts.fs, opts.DockerfilePath); err != nil {
+		isDir, err := afero.IsDir(opts.fs, opts.DockerfilePath)
+		if err != nil {
 			return err
 		}
+		if isDir {
+			return fmt.Errorf("dockerfile path expected, got %s", opts.DockerfilePath)
```

Comment:
nit: This error message doesn't mention that the path is a directory, maybe "Dockerfile path is a directory:%s, please provide path to file."